weave

```diff
@@ -1177,6 +1181,17 @@ launch_router() {
         -e WEAVE_PASSWORD \
         -e WEAVE_CIDR=none \
         $WEAVE_DOCKER_ARGS $IMAGE $COVERAGE_ARGS --iface $CONTAINER_IFNAME --port $CONTAINER_PORT --name "$PEERNAME" --nickname "$(hostname)" --ipalloc-range "$IPRANGE" --dns-effective-listen-address "$DOCKER_BRIDGE_IP" "$@")
+}
+
+# Recreate the parameter values that are set when the router is first launched
+fetch_router_args() {
+    IPRANGE=$(docker inspect -f '{{.Args}}' $1 | grep -o -e '-ipalloc-range [^ ]*')
```

Comment:
The args are dumped as a sequence of name/value pairs, all space-separated.  I don't see that an extra dash would break it, if that's what you meant.

Since, in this code, we only need the presence/absence of `IPRANGE` and `DNS_PORT_MAPPING`, it is enough.  Having the script call an http interface on the router would also be fine, but would need to be implemented.

Currently, the proxy does not call the router at all; that seems a step too far to introduce here.